channeldb/migration_01_to_11/channel_cache_test.go

```diff
@@ -0,0 +1,105 @@
+package migration_01_to_11
+
+import (
+	"reflect"
+	"testing"
+)
+
+// TestChannelCache checks the behavior of the channelCache with respect to
+// insertion, eviction, and removal of cache entries.
+func TestChannelCache(t *testing.T) {
```

Comment:
Do we really need to copy over all the tests as well? Can't we just copy the call graph of any encode/decode functions used in the migration.

Reply:
Previously I found that I needed some test code because it was reused. Left all of it in, but I think that also had a lot of deps that prevents more removal of code. Will look at the diff again.

Reply:
Ok, removed many more lines. Down to 5k lines now.

I used `staticcheck` from https://github.com/dominikh/go-tools with the `-unused.whole-program` flag to find more unused code.

Reply:
Ok, removed many more lines. Down to 5k lines now.

I used `staticcheck` from https://github.com/dominikh/go-tools with the `-unused.whole-program` flag to find more unused code.